homeassistant/components/nextdns/__init__.py

```diff
@@ -0,0 +1,185 @@
+        except (ApiError, ClientConnectorError, InvalidApiKeyError) as err:
+            raise UpdateFailed(err) from err
+
+
+class NextDnsEncryptionUpdateCoordinator(NextDnsUpdateCoordinator):
+    """Class to manage fetching NextDNS analytics encryption data from API."""
+
+    async def _async_update_data(self) -> AnalyticsEncryption:
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                return await self.nextdns.get_analytics_encryption(self.profile_id)
+        except (ApiError, ClientConnectorError, InvalidApiKeyError) as err:
+            raise UpdateFailed(err) from err
+
+
+class NextDnsIpVersionsUpdateCoordinator(NextDnsUpdateCoordinator):
+    """Class to manage fetching NextDNS analytics IP versions data from API."""
+
+    async def _async_update_data(self) -> AnalyticsIpVersions:
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                return await self.nextdns.get_analytics_ip_versions(self.profile_id)
+        except (ApiError, ClientConnectorError, InvalidApiKeyError) as err:
+            raise UpdateFailed(err) from err
+
+
+class NextDnsProtocolsUpdateCoordinator(NextDnsUpdateCoordinator):
+    """Class to manage fetching NextDNS analytics protocols data from API."""
+
+    async def _async_update_data(self) -> AnalyticsProtocols:
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                return await self.nextdns.get_analytics_protocols(self.profile_id)
+        except (ApiError, ClientConnectorError, InvalidApiKeyError) as err:
+            raise UpdateFailed(err) from err
+
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+COORDINATORS = [
+    (ATTR_DNSSEC, NextDnsDnssecUpdateCoordinator, UPDATE_INTERVAL_ANALYTICS),
+    (ATTR_ENCRYPTION, NextDnsEncryptionUpdateCoordinator, UPDATE_INTERVAL_ANALYTICS),
+    (ATTR_IP_VERSIONS, NextDnsIpVersionsUpdateCoordinator, UPDATE_INTERVAL_ANALYTICS),
+    (ATTR_PROTOCOLS, NextDnsProtocolsUpdateCoordinator, UPDATE_INTERVAL_ANALYTICS),
+    (ATTR_STATUS, NextDnsStatusUpdateCoordinator, UPDATE_INTERVAL_ANALYTICS),
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up NextDNS as config entry."""
+    api_key = entry.data[CONF_API_KEY]
+    profile_id = entry.data[CONF_PROFILE_ID]
+
+    websession = async_get_clientsession(hass)
+    try:
+        async with timeout(10):
+            nextdns = await NextDns.create(websession, api_key)
+    except (ApiError, ClientConnectorError, asyncio.TimeoutError) as err:
+        raise ConfigEntryNotReady from err
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN].setdefault(entry.entry_id, {})
+
+    tasks = []
+
+    # Independent DataUpdateCoordinator is used for each API endpoint to avoid
+    # unnecessary requests when entities using this endpoint are disabled.
```

Comment:
Wouldn't they register and start updating anyhow or am I missing something?

Reply:
All coordinators are registered and update data during HA startup. After that, only coordinators with enabled entities will update the data.

Reply:
Really. Need to remember that.